src/python/WMCore/REST/Validation.py

```diff
@@ -35,11 +37,17 @@ def _check_str(argname, val, rx, custom_err = None):
     """
     if isinstance(val, str):
         try:
-            val = bytes(val, "utf-8")
+            val = val.encode("utf-8")
         except:
             raise InvalidParameter(return_message("Invalid '%s' parameter %s %s" % (argname, val, type(val)), custom_err))
-    if not isinstance(val, basestring) or not rx.match(val):
+    if not isinstance(val, basestring):
         raise InvalidParameter(return_message("Incorrect '%s' parameter" % argname, custom_err))
+    try:
+        if not rx.match(val):
+            raise InvalidParameter(return_message("Incorrect '%s' parameter" % argname, custom_err))
+    except TypeError:  # if we match bytes against a string pattern
+        if not rx.match(decodeBytesToUnicode(val)):
+            raise InvalidParameter(return_message("Incorrect '%s' parameter" % argname, custom_err))
```

Comment:
The problem that we try to solve here is: sometimes `val` is bytes, but `rx` is a regex pattern compiled from unicode. Since i could not find a way of programmatically knowing the type of the string used to compile the pattern, what i lazily propose here is:

assumptions
- if `val` is unicode, we convert it to bytes
- if `val` is not bytes, we raise InvalidParameter

options, knowing that `val` is bytes:
1. if rx is a pattern from bytes, then `rx.match(val)` returns a boolean correctly. if this is not True, then we raise an InvalidParameter exception.
2. if rx is a pattern from unicode, `rx.match(val)` raises a TypeError exception, which is catched below by `except TypeError`. now `rx.match(decodeBytesToUnicode(val))` returns a valid boolean. if this is not True, then we raise and InvalidParameter exception.

It is possible that there is a simpler and better way to deal with this, i am open to suggestions!

When we want to check if a bytes string matches a unicode pattern, i assume that the bytes string can be safely decoded to unicode. This is not true in general (one may have a bytes stream containing data from an image or root file), but i assume that in this case we wouldn't run `validate_str()` on such a stream in py2. Or at least, we would pass a pattern that is explicitly of bytes type.

Reply:
Dario, sorry for suggesting another approach, but what if we enforce `val` to be a py2/py3 string object instead of bytes? You said the regex `rx` is a py3 string, which makes me think that we should actually expect a py3 string `val` as well. I could very well be missing something here.

Reply:
your suggestion seems the most reasonable one to me! i do not know-remember why i did not go for it, let me try!

Reply:
Here we go: `convert unicode to str (bytes) first to support cherrypy 3.2.2` is in the docstring, this is why i did not go for that route

Reply:
I couldn't find anything clear on the CP documentation, besides some bug fixes for response header and body. 
We are now in version 17.4.0, if you think it's worth trying the string object approach, I'm all for it. In the end, whatever approach we take here, a reasonable "real" test will be required.

Reply:
I am currently trying to see what happens if I assume that `val` and `rx` are unicode. If this works, then `_check_ustr()`/`validate_ustr()` become redundant. They do not seem to be used anywhere by WMCore, but maybe someone else could be relying on those.